### DISCUSSION TITLE
How to use Custom Rule in Rule Attributes

### DISCUSSION THREAD

**Discussion by sohailahmad07** [livewire/livewire#6634] [4. General]
Hi i have a custom rule that accepts a value but I can't use it in the Rule attribute i throw the below error maybe i am doing wrong i don't know  
```
#[Rule('required')]
public string $type = '';

#[Rule(new ValidMediaFile($this->type))]
public UploadedFile|string|null $body = null;
```
error
```
Constant expression contains invalid operations
```

right now i revert back to using the rules method

**Comment by jj15asmr** [comment]
From the v3 [docs on validation](https://livewire.laravel.com/docs/validation#rule-attributes): 

> PHP Attributes are restricted to certain syntaxes like plain strings and arrays. If you find yourself wanting to use run-time syntaxes like Laravel's rule objects (Rule::exists(...)) you should instead define a rules() method in your component.

So using a rule object will require the usage of the ``rules()`` method. :)
